### PR TITLE
Add initial Dockerfile

### LINKED ISSUE #87: Write a Dockerfile to build the image to host docs.ubuntu.com
The Dockerfile should be based on [the www.ubuntu.com Dockerfile](https://github.com/canonical-websites/www.ubuntu.com/blob/master/Dockerfile), including:

- Not perform any building of assets - this should be done beforehand
- Copy only the necessary files into the image from this project
- Running the app with python 3 (fixes #64)
- Be based on `ubuntu:xenial`
- Install any necessary *production* dependencies into the image
- Take a `commit_id` argument and create `COMMIT_ID` env var from it (https://github.com/canonical-websites/www.ubuntu.com/blob/master/Dockerfile#L12)
- Run this site as a webserver on port 80 as its default action

### PR DESCRIPTION
**Pull request by WillMoggridge**
- Add Dockerfile for building web app image

Fixes #87 

QA
--

Build and run the docker image:

``` bash
docker build --tag docs.ubuntu.com --build-arg COMMIT_ID=`git rev-parse HEAD` .
docker run --rm -p 80:80 docs.ubuntu.com
```

Go to <http://127.0.0.1> and see the docs site

### CODE REVIEW

**Comment by nottrobin** [comment]
@WillMoggridge I've made #87, and it defines some specifics of how this Dockerfile should be created.

Could you update these Dockerfiles based on the points in there. so that it fixes #87 ?

**Comment by WillMoggridge** [comment]
Hey @nottrobin I updated the build to be more inline with the www.ubuntu.com one.

I'm not sure what to do about the assets build at this moment because if you run `./run build`, it builds all the docs which we don't want to add to the base image.
Should I remove it anyway and we work that out the details later, or should I leave it in there for now?

**Comment by nottrobin** [comment]
@WillMoggridge At present, that is the shape of the docs.ubuntu.com app - with all the documentation. So could we just build the image with all the documentation for now?

**Comment by WillMoggridge** [comment]
I have updated the PR to reflect that.

**Comment by WillMoggridge** [comment]
Fixes #87

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# System dependencies
+RUN apt-get update && apt-get install --yes python3-pip
+
+# Python dependencies
+ENV LANG C.UTF-8
+RUN pip3 install --upgrade pip && pip3 install gunicorn -r requirements.txt
+
+# Set git commit ID
+ARG commit_id
```
Comment (nottrobin) [inline Dockerfile]: After writing it this way, I now think it's better to do simply:

```
ARG COMMIT_ID=""
```

Firstly, I discovered there's no need to declare the argument as a proper environment variable - it works just the same without. Secondly, if we give it a default value then this repo can be build without the argument, e.g. for auto-building with hub.docker.com.

Comment (WillMoggridge) [reply Dockerfile]: I have updated this now

**Pull request merged**